migrations/versions/0313_add_last_validated_at.py

```diff
@@ -0,0 +1,43 @@
+"""
+
+Revision ID: 0313_add_last_validated_at
+Revises: 0312_populate_returned_letters
+Create Date: 2020-01-28 18:03:22.237386
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = '0313_add_last_validated_at'
+down_revision = '0312_populate_returned_letters'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('users', sa.Column('email_access_last_validated_at', sa.DateTime(), nullable=True))
+    # if user has email_auth, set email_access_last_validated_at on last login, else set it at user created_at date.
```

Comment:
Do we record anything in the DB that would indicate a user using sms auth reset their password at a particular time (as the resetting of a password does require email access)? 

If not, then this migration looks good

Reply:
We do not record what way password has been changed, so we have no way of knowing retroactively if a user had to use their email to change the password or not.